File: src/main/java/com/example/SlotlyV2/feature/schedule/BlockedPeriod.java
package com.example.SlotlyV2.feature.schedule;

import java.time.OffsetDateTime;

import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class BlockedPeriod {
    private OffsetDateTime startTime;
    private OffsetDateTime endTime;
    private boolean isRecurring;
}

File: frontend/src/features/schedule/types/BlockedPeriod.ts
export interface BlockedPeriodResponse {
    id: string;
    startTime: string;
    endTime: string;
    reason?: string;
    isRecurring: string;
}

export interface BlockedPeriodRequest {
    startTime: string;
    endTime: string;
    reason?: string;
    isRecurring: string;
}
